File: macos/FivePad/Theme/Tokens.swift
import SwiftUI

/// Titik pendaratan nilai desain, kembaran `Tokens.kt` di Android.
///
/// Angka Figma dipakai apa adanya: bingkai 375 pt di Figma = 375 pt di macOS,
/// sama seperti 375 dp di Android. Yang berubah antar platform hanya tata
/// letaknya, bukan ukurannya.
enum Tokens {
    static let space1: CGFloat = 4
    static let space2: CGFloat = 8
    static let space3: CGFloat = 12
    static let space4: CGFloat = 16
    static let space5: CGFloat = 20
    static let space6: CGFloat = 24

    /// Sisi kiri-kanan isi catatan.
    static let screenPadding = space4

    static let radiusSm: CGFloat = 12
    static let radiusMd: CGFloat = 16
    static let radiusLg: CGFloat = 18
    static let radiusPill: CGFloat = 999

    // Bilah atas
    static let topBarHeight: CGFloat = 56
    static let dot: CGFloat = 24
    static let dotRing: CGFloat = 2
    static let dotGap = space4
    static let titleRowHeight: CGFloat = 40
    static let stripeHeight: CGFloat = 4

    // Bilah bawah
    static let navHeight: CGFloat = 62
    static let pillWidth: CGFloat = 84
    static let pillHeight: CGFloat = 52

    /// Area sentuh/klik minimum yang wajar — titik slot 24 pt jauh di
    /// bawahnya, jadi sasaran kliknya diperbesar tanpa mengubah ukuran
    /// lingkaran yang tampak.
    static let touchTarget: CGFloat = 48

    static let bodyTextSize: CGFloat = 16
    static let bodyLineHeight: CGFloat = 28
    static let captionTextSize: CGFloat = 12

    /// FR-5.1: jendela tidak boleh lebih kecil dari ini.
    static let minWindow = CGSize(width: 480, height: 420)

    /// Di bawah lebar ini, tata letaknya runtuh jadi satu kolom bertab —
    /// persis layar Android. Di atasnya, catatan dan tugas berdampingan.
    static let sideBySideWidth: CGFloat = 720
}

File: android/app/src/main/java/com/fivepad/app/ui/theme/Tokens.kt
package com.fivepad.app.ui.theme

import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/**
 * Titik pendaratan nilai desain.
 *
 * Semua jarak, radius, dan ukuran hidup di sini, bukan tersebar sebagai angka
 * ajaib di dalam composable. Skala empat titik dipertahankan, lalu radius dan
 * tinggi kontrol mengikuti bahasa visual gelap dan lembut aplikasi.
 */
object Tokens {

    /** Skala spasi. Kelipatan 4, jadi semua jarak tetap sebidang. */
    val space1 = 4.dp
    val space2 = 8.dp
    val space3 = 12.dp
    val space4 = 16.dp
    val space5 = 20.dp
    val space6 = 24.dp

    /** Sisi kiri-kanan isi catatan. */
    val screenPadding = space4

    val radiusSm = 12.dp
    val radiusMd = 16.dp
    val radiusLg = 18.dp
    val radiusPill = 999.dp

    // ---- Bilah atas catatan ----

    /** Tinggi bilah atas, dan lebar kotak ikon di kedua ujungnya. */
    val topBarHeight = 56.dp

    /** Titik penanda slot: 24 dp, dengan cincin putih 2 dp pada yang aktif. */
    val dot = 24.dp
    val dotRing = 2.dp
    val dotGap = space4

    /** Baris judul catatan memberi ruang untuk gaya nama 17 sp. */
    val titleRowHeight = 40.dp

    /** Pita warna slot di bawah baris judul. */
    val stripeHeight = 4.dp

    // ---- Bilah bawah ----

    val navHeight = 62.dp
    val pillWidth = 84.dp
    val pillHeight = 52.dp

    /**
     * Area sentuh minimum. Pedoman Material meminta 48 dp; titik yang hanya
     * 24 dp jauh di bawah itu, jadi sasaran sentuhnya diperbesar tanpa
     * mengubah ukuran lingkaran yang tampak.
     */
    val touchTarget = 48.dp

    val bodyTextSize = 16.sp

    /** Comfortable note leading shared by source and formatted views. */
    val bodyLineHeight = 28.sp
    val captionTextSize = 12.sp
}
